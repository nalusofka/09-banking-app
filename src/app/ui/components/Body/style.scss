@use '../../styles/variables' as var;

.body {
  color: var.$color-white;
  line-height: 1.6;
  margin: 0;
}