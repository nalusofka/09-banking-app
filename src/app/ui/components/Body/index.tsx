import React from 'react';
import './style.scss';

interface BodyProps {
  children: React.ReactNode;
  className?: string;
}

const Body: React.FC<BodyProps> = ({ children, className }) => {
  return <p data-testid="body-component" className={className || 'body'}>{children}</p>;
};

export default Body;
