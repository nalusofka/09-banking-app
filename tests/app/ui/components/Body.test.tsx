import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Body from '../../../../src/app/ui/components/Body';

describe('Componente Body', () => {
  it('renderiza correctamente con el contenido proporcionado', () => {
    const { asFragment } = render(<Body>Contenido de prueba</Body>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('aplica la clase correcta según el color primario', () => {
    const { getByTestId } = render(<Body color="primary">Contenido primario</Body>);
    const bodyElement = getByTestId('body-component');
    expect(bodyElement).toHaveClass('body-primary');
  });

  it('aplica la clase correcta según el color secundario', () => {
    const { getByTestId } = render(<Body color="secondary">Contenido secundario</Body>);
    const bodyElement = getByTestId('body-component');
    expect(bodyElement).toHaveClass('body-secondary');
  });

  it('aplica una clase personalizada cuando se proporciona', () => {
    const { getByTestId } = render(
      <Body className="custom-class">Contenido personalizado</Body>
    );
    const bodyElement = getByTestId('body-component');
    expect(bodyElement).toHaveClass('custom-class');
  });

  it('renderiza correctamente con una clase combinada', () => {
    const { getByTestId } = render(
      <Body color="primary" className="custom-class">
        Contenido combinado
      </Body>
    );
    const bodyElement = getByTestId('body-component');
    expect(bodyElement).toHaveClass('custom-class');
    expect(bodyElement).not.toHaveClass('body-primary');
  });
});
