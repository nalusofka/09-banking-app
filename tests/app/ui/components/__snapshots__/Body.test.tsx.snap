// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Body > renderiza correctamente con el contenido proporcionado 1`] = `
<DocumentFragment>
  <p
    class="body' body-secondary"
    data-testid="body-component"
  >
    Contenido de prueba
  </p>
</DocumentFragment>
`;
